src/algebra/algebra/basic.lean

```diff
@@ -1404,20 +1404,40 @@ end linear_map
 end is_scalar_tower
 
 section restrict_scalars
-/- In this section, we describe restriction of scalars: if `S` is an algebra over `R`, then
-`S`-modules are also `R`-modules. -/
 
 section type_synonym
 variables (R A M : Type*)
 
-/--
-Warning: use this type synonym judiciously!
-The preferred way of working with an `A`-module `M` as `R`-module (where `A` is an `R`-algebra),
-is by `[module R M] [module A M] [is_scalar_tower R A M]`.
+/-- A type synonym representing the functor from the category of modules over `A`, regarded as a
```

Comment:
This sentence still contains the (to me) confusing statement... I think it would already help if you replace `functor` be `equivalence of categories`. And also, talking about representations instead of modules, might help.

Reply:
Sorry, it was sloppy of me to forget about this opening remark. I've just pushed some new wording which hopefully helps.

I should say, I don't mind we give up on this PR: I'm not sure I'm helping!